pkg/integrations/prometheus/promql/parser.go

```diff
@@ -607,6 +608,29 @@ func handleVectorSelector(mQueryReqs []*structs.MetricsQueryRequest, intervalSec
 	return mQueryReqs, nil
 }
 
+// handleNumberLiteral
+func handleNumberLiteral(e *parser.NumberLiteral, mQueryReqs []*structs.MetricsQueryRequest,
+	queryArithmetic []*structs.QueryArithmetic) ([]*structs.MetricsQueryRequest, []*structs.QueryArithmetic, error) {
+
+	arithmeticOperation := structs.QueryArithmetic{}
+	var lhsRequest []*structs.MetricsQueryRequest
+	arithmeticOperation.ConstantOp = true
+	arithmeticOperation.Constant = e.Val
+
+	queryArithmetic = append(queryArithmetic, &arithmeticOperation)
+
+	if mQueryReqs[0].MetricsQuery.MQueryAggs == nil {
+		mQueryReqs = lhsRequest
+	} else if mQueryReqs[0].MetricsQuery.HashedMName == uint64(0) {
+		// This means there is a common group by on multiple metrics separated by operators.
+		// So, we need to append the aggregations to each of the Request in LHS and RHS.
+		// And we can discard the current aggregation in the mQueryReqs[0]
+		mQueryReqs = appendMetricAggsToTheMQuery(lhsRequest, &mQueryReqs[0].MetricsQuery)
+	} else {
+		mQueryReqs = append(mQueryReqs, lhsRequest...)
+	}
+	return mQueryReqs, queryArithmetic, nil
+}
```

Comment:
I do not understand what is the purpose of this function. And based on the way the parser is written. We do not want to do this. As this will cause issues with `BinaryExpr` and `CallExpr`.